terraform/examples/cicd_account/run_terraform.sh

```diff
@@ -2,7 +2,7 @@
 
 DELETE_ALL=${1}
 
-if ( [[ -z ${AWS_ACCESS_KEY_ID} ]] || [[ -z ${AWS_SECRET_ACCESS_KEY} ]] ) && [[ -z ${AWS_PROFILE} ]]; then
+if ([[ -z ${AWS_CONTAINER_AUTHORIZATION_TOKEN} ]] && (( [[ -z ${AWS_ACCESS_KEY_ID} ]] || [[ -z ${AWS_SECRET_ACCESS_KEY} ]] ) && [[ -z ${AWS_PROFILE} ]])); then
```

Comment:
This change would be specific for container technology and especially for IoT GreenGrass: https://docs.aws.amazon.com/greengrass/v2/developerguide/run-docker-container.html